### ISSUE TITLE
On books page, for borrowed book, show time remaining on loan

### ISSUE THREAD

**Issue by mekarpeles** [internetarchive/openlibrary#1110]
https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/type/edition/view.html is the file that renders the books page.

The read buttons themselves are rendered using: 
https://github.com/internetarchive/openlibrary/blob/master/openlibrary/macros/LoanStatus.html

The loan/lending logic can be found in:
https://github.com/internetarchive/openlibrary/blob/master/openlibrary/core/lending.py

The **desired outcome** is near the button, similar to the Loans page, we show the expiration date of the loan
<img width="216" alt="Screenshot 2024-03-08 at 12 29 34 PM" src="https://github.com/internetarchive/openlibrary/assets/978325/24d40f11-b237-40dc-be88-1f310ae46f3f">

**Comment by mekarpeles** [comment]
This data is already available on the https://openlibrary.org/account/loans page, it should also show up on the book pages (i.e. work + edition page), e.g. https://openlibrary.org/works/OL14909099W/Debt / https://openlibrary.org/books/OL24088724M/Debt

**Comment by tabshaikh** [comment]
@mekarpeles maybe you could highlight some files to work on in this issue

**Comment by swastikbanerjee07** [comment]
Hi ! I'm just beginning with open-source. Can you guide me how to work on this issue?

**Comment by mekarpeles** [comment]
Hi @swastikbanerjee07!
1) First, follow the instructions to setup Open Library using Docker
2) You may want to borrow a book on production (https://openlibrary.org) and then navigate to your /account/loans page. The corresponding code should be in `openlibrary/templates/account/borrow.html` You should see a block of code in the template like `loan['expiry']` which is how that template is getting the loan expiration date
3) On the books page template (`openlibrary/templates/type/edition/view.html`) you'll see see a section called `editionTools` which loads a macro `$:macros.databarWork(page, editions_page=True)`. The code to this macro is in `openlibrary/macros/databarWork.html`
4. Within the databarWork.html macro, you'll have to fetch a loan (if it exists) just like is done in the `loans` page in step #2. And then you'll want to display the time remaining. Once you're at this step, we'll be able to help guide you regarding the style / position, etc.

One thing which makes this task a bit hard is loans don't really work on local developer instances. We can help you mock them or, once you have something you think is working, we can run the code on https://dev.openlibrary.org so you can test against real data.

**Comment by Zacharius** [comment]
I have a likely solution at https://github.com/Zacharius/openlibrary/tree/1110/feature/loan-timer-on-book-page. Can someone run it on the dev site to see if it works?

I ran the docker image with the code and it didnt give any errors or look wonky on the books page so thats promising. Not sure how else to test it

**Comment by Zacharius** [comment]
@mekarpeles any chance you can help me test this?

**Comment by jdlrobson** [comment]
^ ping @mekarpeles can you give @Zacharius some feedback?

**Comment by shreyansh23** [comment]
@mekarpeles I was trying to solve this issue.
I followed the instructions in your [previous comment](https://github.com/internetarchive/openlibrary/issues/1110#issuecomment-428537080). 
I observed how expiry data is coming in /account/loans page. 
I reached here,

```class account_loans(delegate.page):
    path = "/account/loans"

    @require_login
    def GET(self):
        user = accounts.get_current_user()
        user.update_loan_status()
        loans = borrow.get_loans(user)
        return render['account/borrow'](user, loans)
```


I want this `loans` object available to edition templates. I am stuck how to implement that. I think I am thinking in the wrong way. 
Any suggesions or can I get expiry date in some other way?
